websites/R/R-a-d.io/presence.ts

```diff
@@ -0,0 +1,94 @@
+const presence = new Presence({
+		clientId: "1073136952905318401",
+	}),
+	/**
+	 * Returns current player status
+	 * @returns {string} either "Stopped", "Playing", or "Connecting..."
+	 */
+	playStatus = (): string => {
+		let status: string;
+		const playButtonText: string =
+			document.querySelector("#stream-play").textContent;
+		if (playButtonText === "Play Stream") status = "Stopped";
+		else if (playButtonText === "Stop Stream") status = "Playing";
+		else status = playButtonText;
+		return status;
+	},
+	/**
+	 * Returns current song playing
+	 * @returns {string} Current song, usually in "{artist} - {title}" format
+	 */
+	currentSong = (): string => {
+		return document.querySelector("#np").textContent;
+	};
+
+presence.on("UpdateData", async () => {
+	const path = document.location.pathname,
+		presenceData: PresenceData = {
+			// R/a/dio logo
+			largeImageKey: "https://i.imgur.com/o6JjbtH.png",
+		};
+	if (path.includes("/news")) {
+		// On News page
+		presenceData.details = "Reading news";
+		// Current article title if currently being open
+		if (path.split("/").length === 3) {
+			presenceData.state = document.querySelector(
+				'[class="panel-title"] > a'
+			).textContent;
+		}
+	} else if (path.includes("/irc")) {
+		// On Chat page
+		presenceData.details = "Chatting";
+	} else if (path.includes("/search")) {
+		// On Search page
+		const query = path.split("/");
+		// If searching for something
+		if (query[2]) presenceData.details = `Searching for: ${query[2]}`;
```

Comment:
```suggestion
		const title = path.split("/")[2];
		// If searching for something
		if (title) presenceData.details = `Searching for: ${title}`;
```

Something like this could improve readability